### ISSUE TITLE
Installing on an Ubuntu 18.04 server from the repo, boundary dev will not let login occur due to missing service.

### ISSUE THREAD

**Issue by birdiesanders** [hashicorp/boundary#696]
**Describe the bug**

**To Reproduce**
.5. Install the ubuntu package for boundary and start the dev env.
1. Run `boundary authenticate password -login-name=admin -password password -auth-method-id=ampw_123456789`
2. See error

**Expected behavior**
I expect to login. 

**Additional context**
Error as reported: 

Error reading auth token from system credential store: The name org.freedesktop.secrets was not provided by any .service files
Error from controller when performing authentication: 
Error information:
  Code:                NotFound
  Message:             Resource not found.
  Status:              404

**Comment by saiyam1814** [comment]
Can confirm it happens on centos7 as well

**Comment by jefferai** [comment]
You're missing a zero at the end of the auth method id -- `ampw_1234567890`!

**Comment by jefferai** [comment]
Also if you're running the client on an environment that does not have software installed that provides `org.freedesktop.secrets`, you can pass `-token-name=none` to turn off credential storing and lookup.

**Comment by saiyam1814** [comment]
@jefferai  SO I did install few things to make that command work 
```
boundary authenticate password -auth-method-id=ampw_1234567890 \
>     -login-name=admin -password=password
Error reading auth token from system credential store: exec: "dbus-launch": executable file not found in $PATH

Authentication information:
  Account ID:      apw_EJK7s6DcCQ
  Auth Method ID:  ampw_1234567890
  Expiration Time: Thu, 22 Oct 2020 10:17:30 BST
  Token:
  at_h1qSNMdB1c_s125wLmVW7LgzNKf1TFeNrFDRUZNcR2qeWkNyjZ7Qd6E2DWGsyPT9KmPKZkxBaQps7JKkbeeoJEwt7xXMyR6YKjEkqbrFWcsCdXm8rYgqjsFJwUYS2WBFbNh2
  User ID:         u_1234567890
Error saving auth token to system credential store: exec: "dbus-launch": executable file not found in $PATH

```


Installed dbus 
```
apt install dbus-x11
```

```
boundary authenticate password -auth-method-id=ampw_1234567890     -login-name=admin -password=password
Error reading auth token from system credential store: The name org.freedesktop.secrets was not provided by any .service files

Authentication information:
  Account ID:      apw_EJK7s6DcCQ
  Auth Method ID:  ampw_1234567890
  Expiration Time: Thu, 22 Oct 2020 10:19:18 BST
  Token:
  at_VqoR4tBhxy_s18pG3buSH7fMcBVL1YP14NXipTgeg4Jm9aVd3WXL2WCpUvqs8xnCwnNC7D41zejxUfRYhJPGMvCcp1mbN1q9cPyM5AAJRvakFGQUNLC5ZjiyyoTuic4ZpoQUmu9XXtC2YY2u9BTdJDg8
  User ID:         u_1234567890
Error saving auth token to system credential store: The name org.freedesktop.secrets was not provided by any .service files
```

Next installed 
```
sudo apt-get install -y gnome-keyring
```

```
boundary authenticate password -auth-method-id=ampw_1234567890     -login-name=admin -password=password
Error reading auth token from system credential store: failed to unlock correct collection '/org/freedesktop/secrets/aliases/default'

Authentication information:
  Account ID:      apw_EJK7s6DcCQ
  Auth Method ID:  ampw_1234567890
  Expiration Time: Thu, 22 Oct 2020 10:21:00 BST
  Token:
  at_7y54bPBIDx_s13PefhTbtKamPce8iph3t8HMK85nJWEBh9n2JXCk6oiWQ9K9qusfJDx6TEJPLoGo8GPqbawpLAxtMgk9aS5wyzr3S6qVgMG7S939K93pUXLffE7a5KXdwAc464p42rV2
  User ID:         u_1234567890
Error saving auth token to system credential store: failed to unlock correct collection '/org/freedesktop/secrets/aliases/default'
```

Am I doing totally wrong steps or I can only run with --token-name=none?

**Comment by jefferai** [comment]
You might need to log out/in (or even reboot) after installing those in order for things to be set up in your account correctly.

Everything is still usable -- you would simply need to pass that token via `BOUNDARY_TOKEN` or the `-token` flag without the OS credential store. `token-name=none` will just remove the errors you're seeing, but will also not cause the token to be saved.

Apologies for the confusion here, we'd like to handle it better and thinking about how to do it, but there's a tradeoff between silencing those errors where the user may get confused as to why a token isn't being stored (or then read back out) and passing them through.

**Comment by birdiesanders** [comment]
I can confirm passing the token does work, inconvenient, but functional.

**Comment by jefferai** [comment]
I'm going to close this as the original issue (the 404) is solved, and the OP has opened #697 around issues setting up the credential store on Ubuntu.

@birdiesanders Agreed -- we thought about a Vault-style storing of the token in a local file as a backup method but over time we've had complaints around that approach from people that didn't realize it was being stored in plaintext on disk. Since it's relatively easy to make an alias that will store the token in a file for you on authentication and read it on other commands, or to send the token to some other preferred secrets mechanism we decided to leave it up to the user for the moment if the built-in integration isn't working for them or is disabled.
